src/app/pages/users/user/stripe/payment-methods/payment-methods-table-data-source.ts

```diff
@@ -0,0 +1,233 @@
+          observer.error(error);
+        });
+      } else {
+        observer.next({
+          count: 0,
+          result: [],
+        });
+        observer.complete();
+      }
+    });
+  }
+
+  public buildTableDef(): TableDef {
+    return {
+      search: {
+        enabled: false,
+      },
+      hasDynamicRowAction: true,
+    };
+  }
+
+  public buildTableColumnDefs(): TableColumnDef[] {
+    const columns: TableColumnDef[] = [
+      {
+        id: 'status',
+        name: 'general.status',
+        isAngularComponent: true,
+        angularComponent: PaymentMethodStatusComponent,
+        headerClass: 'text-center col-15p',
+        class: 'text-center col-15p',
+      },
+      {
+        id: 'default',
+        name: 'general.default',
+        headerClass: 'text-center col-10p',
+        class: 'text-center col-10p',
+        formatter: (defaultPaymentMethod: boolean, paymentMethod: BillingPaymentMethod) => {
+          return paymentMethod.isDefault ?
+            this.translateService.instant('general.yes') : this.translateService.instant('general.no');
+        },
+      },
+    ]
+      if (!this.isAdmin) {
+        columns.push(
+          {
+            id: 'id',
+            name: 'general.id',
+            headerClass: 'text-center col-10p',
+            class: 'text-center col-10p',    
+          },
+        );
+      }
+      columns.push(
+        {
+          id: 'type',
+          name: 'settings.billing.payment_methods_type',
+          headerClass: 'text-center col-10p',
+          class: 'text-center col-10p capitalize',
+          },
+        {
+          id: 'brand',
+          name: 'settings.billing.payment_methods_brand',
+          headerClass: 'text-center col-15p',
+          class: 'text-center col-15p capitalize',
+          },
+        {
+          id: 'last4',
+          name: 'settings.billing.payment_methods_ending_with',
+          headerClass: 'text-center col-10p',
+          class: 'text-center col-10p',
+          },
+        {
+          id: 'expiringOn',
+          name: 'settings.billing.payment_methods_expiring_on',
+          headerClass: 'text-center col-10p',
+          class: 'text-center col-10p',
+          formatter: (expiringOn: Date) => {
+            return this.datePipe.transform(expiringOn, 'MM/YYYY');
+          },
+        },
+        {
+          id: 'createdOn',
+          name: 'general.created_on',
+          headerClass: 'text-center col-15p',
+          class: 'text-center col-15p',
+          formatter: (createdOn: Date) => this.datePipe.transform(createdOn)
+        },
+      )
+    return columns;
+  }
+
+  public buildTableActionsDef(): TableActionDef[] {
+    const tableActionsDef = super.buildTableActionsDef();
+    if (this.activatedRoute.snapshot.url[0]?.path === 'profile') {
+      this.currentUserID = this.centralServerService.getLoggedUser().id;
+    }
+    this.canCreatePaymentMethod = this.authorizationService.canCreatePaymentMethod(this.currentUserID);
```

Comment:
Move this line of code in the constructor

Reply:
will try to make the other changes you provided first because if it's here and not in the constructor it's because I didn't manage to get the currentUserID set at constructor time